api/src/main/java/org/openmrs/util/ConceptReferenceRangeUtility.java

```diff
@@ -0,0 +1,127 @@
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A utility class that evaluates the concept ranges 
+ * 
+ * @since 2.7.0
+ */
+public class ConceptReferenceRangeUtility {
+	
+	private static final Logger logger = LoggerFactory.getLogger(ConceptReferenceRangeUtility.class);
+	
+	public ConceptReferenceRangeUtility() {
+	}
+	
+	/**
+	 * This method evaluates the given criteria against the provided person/patient.
+	 *
+	 * @param criteria the criteria string to evaluate e.g. "$fn.getAge($patient, 'YEARS') > 1 && $fn.getAge($patient, 'YEARS') < 10"
+	 * @param person person object containing attributes to be used in the criteria
+	 *                  
+	 * @return true if the criteria evaluates to true, false otherwise
+	 * 
+	 * @since 2.7.0
+	 */
+	public boolean evaluateCriteria(String criteria, Person person) {
+		if (person == null) {
+			throw new IllegalArgumentException("Failed to evaluate criteria with reason: patient is null");
+		}
+		
+		if (StringUtils.isBlank(criteria)) {
+			throw new IllegalArgumentException("Failed to evaluate criteria with reason: criteria is empty");
+		}
+		
+		VelocityContext velocityContext = new VelocityContext();
+		velocityContext.put("fn", this);
+		
+		velocityContext.put("patient", person);
+		
+		VelocityEngine velocityEngine = new VelocityEngine();
+		
+		StringWriter writer = new StringWriter();
+		String wrappedCriteria = "#set( $criteria = " + criteria + " )$criteria";
+		
+		try {
+			velocityEngine.evaluate(velocityContext, writer, ConceptReferenceRangeUtility.class.getName(), wrappedCriteria);
+			String evaluatedCriteria = writer.toString();
+			return Boolean.parseBoolean(evaluatedCriteria);
+		}
+		catch (ParseErrorException e) {
+			throw new APIException("An error occurred while evaluating criteria: Invalid criteria: " + criteria, e);
+		}
+		catch (Exception e) {
+			throw new APIException("An error occurred while evaluating criteria: ", e);
+		}
+	}
+	
+	/**
+	 * Gets the latest Obs by concept.
+	 *
+	 * @param conceptRef can be either concept uuid or conceptMap's code and sourceName 
+	 *                   e.g "bac25fd5-c143-4e43-bffe-4eb1e7efb6ce" or "CIEL:1434"
+	 *                   
+	 * @return Obs
+	 * 
+	 * @since 2.7.0
+	 */
+	public Obs getLatestObsByConcept(String conceptRef) {
+		Concept concept = Context.getConceptService().getConceptByReference(conceptRef);
+
+		if (concept != null) {
+			List<Obs> observations = Context.getObsService().getObservations(
+				null, 
+				null, 
+				Collections.singletonList(concept), 
+				null, 
+				null, 
+				null,
+				Collections.singletonList("dateCreated"), 
+				1, 
+				null,
+				null, 
+				null, 
+				false
+			);
+
+			// Return the latest Obs by sorting the list by dateCreated
```

Comment:
The above is exactly the problem with comments. They increase the maintenance burden because every time you change the code, you need to update them too. When you forget, you get into an out of sync state of the code and comments, like in this case. 😊